Jobs [Distributed Scheduler] API docs, pt.1 (#4164)
dapr/docs: daprdocs/content/en/reference/api/scheduler_api.md

---
type: docs
title: "Scheduler API reference"
linkTitle: "Scheduler API"
description: "Detailed documentation on the scheduler API"
weight: 1300
---

{{% alert title="Note" color="primary" %}}
Scheduler is currently in alpha.
{{% /alert %}}

With the Scheduler API, you can orchestrate jobs and tasks in your environment.

## Schedule a job

Schedule a job with the given name.

```

> POST http://localhost:3500/v1.0-alpha1/scheduler/job/schedule/<name>

Reviewer: ```suggestion
POST http://localhost:3500/v1.0-alpha1/job/schedule/<name>
```

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

### Request content

Any request content will be passed to the scheduler as input. The Dapr API passes the content as-is without attempting to interpret it.

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed

> `500`  | Request formatted correctly, error in dapr code or underlying component

Reviewer: ```suggestion
`500`  | Request formatted correctly, error in dapr code or Scheduler control plane service
```
The etcd in scheduler isn't a dapr component, maybe we can just say scheduler to encapsulate both scheduler + etcd?

### Response content

The API call will provide a response similar to this:

```json
```

## Get job data

Get a job's data with its given name.

```

> GET http://localhost:3500/v1.0-alpha1/scheduler/job/<name>

Reviewer: ```suggestion
GET http://localhost:3500/v1.0-alpha1/job/<name>
```

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed

> `500`  | Request formatted correctly, error in dapr code or underlying component

Reviewer: same here

### Response content

The API call will provide a response similar to this:

> ```json

Reviewer: Actually, just saw this. Rn, if I schedule/get/delete a job successfully this is what I get:
```
$ curl -X POST \
  http://localhost:3500/v1.0-alpha1/job/schedule/test1 \
  -H "Content-Type: application/json" \
    -H 'dapr-app-id: sub' \
  -d '{
        "job": {
            "data": {
                "@type": "type.googleapis.com/google.type.Expr",
                "expression": "cassie87"
            },
            "dueTime": "30s"
        }
    }'

$ curl -X GET http://localhost:3500/v1.0-alpha1/job/test1 -H "Content-Type: application/json" -H 'dapr-app-id: sub'
{"name":"test1","dueTime":"30s","data":{"@type":"type.googleapis.com/google.type.Expr","expression":"cassie87"}}%                                    

$ curl -X DELETE http://localhost:3500/v1.0-alpha1/job/test1 -H "Content-Type: application/json" -H 'dapr-app-id: sub'
```

Because the return types are `google.protobuf.Empty` for the schedule & delete

```
## Delete a job

Delete a job you've created and scheduled with its given name.

```

> DELETE http://localhost:3500/v1.0-alpha1/scheduler/job/<name>

Reviewer: ```suggestion
DELETE http://localhost:3500/v1.0-alpha1/job/<name>
```

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed

> `500`  | Request formatted correctly, error in dapr code or underlying component

Reviewer: same here

---
type: docs
title: "Scheduler API reference"
linkTitle: "Scheduler API"
description: "Detailed documentation on the scheduler API"
weight: 1300
---

{{% alert title="Note" color="primary" %}}
Scheduler is currently in alpha.
{{% /alert %}}

> With the Scheduler API, you can orchestrate jobs and tasks in your environment.

Reviewer 2: ```suggestion
With the scheduler API, you can schedule jobs and tasks in the future.
```

## Schedule a job

> Schedule a job with the given name.

Reviewer 2: ```suggestion
Schedule a job with a name.
```

```

> POST http://localhost:3500/v1.0-alpha1/job/schedule/<name>

Reviewer 2: This API looks wrong. I would have expected this way.
`POST http://localhost:3500/v1.0-alpha1/schedule/job/<name>`
That is consistent with, for example, invoke/method. This name implies that job should be the name of the API

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

> ### Request content

Reviewer 2: ```suggestion
### Request body
```json
{
  "job": {
      "data": {
          "@type": "type.googleapis.com/google.type.Expr",
          "expression": "cassie87"
      },
      "dueTime": "30s"
  }
}
```

> Any request content will be passed to the scheduler as input. The Dapr API passes the content as-is without attempting to interpret it.

Reviewer 2: ```suggestion
Any request content is passed to the scheduler as input. The Dapr API passes the content as-is without attempting to interpret it.
```

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed
`500`  | Request formatted correctly, error in dapr code or Scheduler control plane service

> ### Response content

Reviewer 2: ```suggestion
### Response body
```

> The API call will provide a response similar to this:

Reviewer 2: ```suggestion
The returned response is empty.
```

> ```json

Reviewer 2: Can you turn this into an example, like has been done with the secrets here with a curl command and this response as the example https://docs.dapr.io/reference/api/secrets_api/#response-with-multiple-secrets-and-multiple-key--values-in-a-secret-eg-kubernetes

> {

Reviewer: This was my POST request, not response. The resp is empty

Reviewer: might be helpful to include the structure for the post request above tho :)

Reviewer 2: @hhunter-ms - I Tried to include the example above in the body. I would also include the the request body can include `duetime` and `data` which should both have a description similar to the request body field here in this example https://docs.dapr.io/reference/api/state_api/#request-body

Reviewer 2: The response should be empty

  "job": {
      "data": {
          "@type": "type.googleapis.com/google.type.Expr",
          "expression": "cassie87"
      },
      "dueTime": "30s"
  }
}
```

## Get job data

> Get a job's data with its given name.

Reviewer 2: ```suggestion
Get a job with its name.
```

```

> GET http://localhost:3500/v1.0-alpha1/job/<name>

Reviewer 2: I missed this, but this should be called

`GET http://localhost:3500/v1.0-alpha1/schedule/job/<name>`

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed
`500`  | Request formatted correctly, error in dapr code or Scheduler control plane service

### Response content

> The API call will provide a response similar to this:

Reviewer 2: ```suggestion
The returned response is JSON containing the `name` of the job, the `dueTime` and  the`data`.
```

> ```json

Reviewer 2: Can you turn this into an example, like has been done with the secrets here with a curl command and this response as the example https://docs.dapr.io/reference/api/secrets_api/#response-with-multiple-secrets-and-multiple-key--values-in-a-secret-eg-kubernetes

{
  "name":"test1",
  "dueTime":"30s",
  "data": {
     "@type":"type.googleapis.com/google.type.Expr",
     "expression":"expression1"
   }
}                                    
```
## Delete a job

> Delete a job you've created and scheduled with its given name.

Reviewer 2: ```suggestion
Delete a named job
```

> Delete a job you've created and scheduled with its given name.

Reviewer 2: ```suggestion
Delete a named job.
```

```

> DELETE http://localhost:3500/v1.0-alpha1/job/<name>

Reviewer 2: This should be
DELETE http://localhost:3500/v1.0-alpha1/schedule/job/<name>

```

### URL parameters

Parameter | Description
--------- | -----------
`name` | Job name

### HTTP response codes

Code | Description
---- | -----------
`202`  | Accepted
`400`  | Request was malformed
`500`  | Request formatted correctly, error in dapr code or Scheduler control plane service

### Response content

None.

> 

Reviewer 2: Same comment here on provide an example with a curl command
